### PR TITLE
DT-97-responsive-homepage

### PR DESCRIPTION
**Pull request by Giang-st1-00**
Responsive homepage

### CODE REVIEW

**Comment by lamxuanthinh** [comment]
@DuyTinh2003  Please review this freature

File: components/common/TextEllipsis/index.tsx [hunk 1]
```diff
@@ -3,30 +3,65 @@ import { useState } from "react";
...
             <p className={`overflow-hidden break-all ${className}`}>
-                {showFullCaption ? (
-                    content
-                ) : (
+                {showFullContent && content}
+                {showFullContent && isHideLess && (
+                    <>
+                        {content + `. `}
+                        <span
+                            onClick={() => {
```
Comment (Duy-Tinh-Dev) [inline components/common/TextEllipsis/index.tsx:47]: i think u should create function in this

File: components/common/TextEllipsis/index.tsx [hunk 2]
```diff
@@ -2,36 +2,66 @@ import { useState } from "react";
...
 }
 
-export default function TextEllipsis({ content, styleContent }: ITextEllipsis) {
-    const MAX_CAPTION_LENGTH = 50;
-    const truncatedCaption =
-        content && content.length > MAX_CAPTION_LENGTH ? content.slice(0, MAX_CAPTION_LENGTH) + "...  " : content;
+export default function TextEllipsis({
+    content,
+    className,
+    MAX_CONTENT_LENGTH = 50,
```
Comment (Duy-Tinh-Dev) [inline components/common/TextEllipsis/index.tsx]: change variable

File: components/common/Comments/CommentForm.tsx [hunk 3]
```diff
@@ -50,14 +50,18 @@ const CommentForm: React.FC<CommentFormProps> = ({
                 )}
             </div>
             <input
-                className="text-black w-[75%] dark:text-white rounded-xl p-3 border border-solid border-green-500 dark:border-white"
+                className="text-black text-[14px] sm:text-[16px] w-[75%] dark:text-white rounded-xl p-3 border border-solid border-green-500 dark:border-white"
                 value={text}
                 onChange={(e: any) => {
```
Comment (Duy-Tinh-Dev) [inline components/common/Comments/CommentForm.tsx]: any

File: components/common/Message/index.tsx [hunk 4]
```diff
@@ -19,6 +19,18 @@ export default function Message({ dataMessage, className, children }: IMessagePr
         setIsMessage(!isMessage);
     };
 
+    useEffect(() => {
+        function handleClickOutside(event: any) {
```
Comment (Duy-Tinh-Dev) [inline components/common/Message/index.tsx]: any

File: components/common/Message/index.tsx [hunk 5]
```diff
@@ -19,6 +19,18 @@ export default function Message({ dataMessage, className, children }: IMessagePr
...
+                setIsMessage(false);
+            }
+        }
+        document.addEventListener("mousedown", handleClickOutside);
+        return () => {
+            document.removeEventListener("mousedown", handleClickOutside);
+        };
+    }, []);
+
     const truncateString = (str: any, num: any) => {
```
Comment (Duy-Tinh-Dev) [inline components/common/Message/index.tsx]: any

File: components/common/Video/DescriptionVideo/index.tsx [hunk 6]
```diff
@@ -11,75 +12,95 @@ interface IDescriptionVideo {
...
+                )}
+
+                {hashtags.length > 0 && (
+                    <div className="flex items-center pb-2">
+                        <div className="flex items-center">
+                            <div>
+                                <FaExternalLinkAlt fontSize={"12px"} />
+                            </div>
+                            <div className="flex">
+                                {hashtags.map((item: any, index: any) => {
```
Comment (Duy-Tinh-Dev) [inline components/common/Video/DescriptionVideo/index.tsx:84]: any

File: components/common/Video/NormalVideo/index.tsx [hunk 7]
```diff
@@ -77,7 +77,7 @@ export default function NormalVideo({ data }: INormalVideo) {
     };
 
     const setTimePlayer = (value: any) => {
```
Comment (Duy-Tinh-Dev) [inline components/common/Video/NormalVideo/index.tsx]: any

File: components/common/Comments/Comments.tsx [hunk 8]
```diff
@@ -8,14 +8,15 @@ import {
...
+import { AiOutlineClose } from "react-icons/ai";
 
 interface IComments {
     currentUserId: string;
     isComment: boolean;
     setComment: Dispatch<SetStateAction<boolean>>;
 }
 
-const Comments: React.FC<IComments> = ({ currentUserId, isComment }: IComments) => {
+const Comments: React.FC<IComments> = ({ currentUserId, isComment, setComment }: IComments) => {
```
Comment (Duy-Tinh-Dev) [inline components/common/Comments/Comments.tsx]: delete React.FC<IComments>

File: components/common/SidebarHeader/index.tsx [hunk 9]
```diff
@@ -1,74 +1,109 @@
...
 import Image from "next/image";
 import Link from "next/link";
-import { BsSearch } from "react-icons/bs";
-import { BiVideoPlus } from "react-icons/bi";
+import { BiLogIn, BiVideoPlus } from "react-icons/bi";
+import Menu from "../Menu";
+import { dataMenuNav } from "../Menu/constants";
+import SearchBar from "../SearchBar";
 
 const SidebarHeader: React.FC<{ active: boolean }> = ({ active }) => {
```
Comment (Duy-Tinh-Dev) [inline components/common/SidebarHeader/index.tsx:10]: React.FC<IComments>

File: components/common/Message/index.tsx (hunk 4 above)
Comment (Giang-st1-00) [reply components/common/Message/index.tsx]: image

File: components/common/Video/DescriptionVideo/index.tsx (hunk 6 above)
Comment (Giang-st1-00) [reply components/common/Video/DescriptionVideo/index.tsx:84]: There is no data for hashtags yet

**Pull request merged**